networks/hamiltonian_net.py

```diff
@@ -5,6 +5,7 @@
 
 import torch
 from torch import nn
+from torch.nn import functional as F
```

Comment:
```suggestion
from torch.nn import functional
```

Reply:
F is not very informative, since we are only gonna  use this in a couple of places I would keep the functional name or something more representative than F

Reply:
Ok that works for me. I used `F` as it looks kind of the standard in pytorch

Reply:
F is used widely in the community, it's ok imo